File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    # no comments...
    def _combinator(self, target, i , path, nums, result):
        if target == 0: 
            result.append(path.copy()) 
            return
        if target < 0: return

        for j in range(i, len(nums)):
            path.append(nums[j])
            self._combinator(target - nums[j], j, path, nums, result)
            path.remove(nums[j])


    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        result, path = [], []
        self._combinator(target, 0, path, nums, result)
        return result

File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
	hm := make(map[int]int)

	for i, num := range nums {
		compl := target - num

		if j, ok := hm[compl]; ok {
			return []int{j, i}
		}
		hm[num] = i
	}

	return nil
}